mono/metadata/class-init.c

```diff
@@ -3846,40 +3844,36 @@ mono_class_layout_fields (MonoClass *klass, int base_instance_size, int packing_
 			real_size = MAX (real_size, size + field_offsets [i]);
 		}
 
-		if (klass->has_references) {
-			ref_bitmap = g_new0 (guint8, real_size / TARGET_SIZEOF_VOID_P);
-
-			/* Check for overlapping reference and non-reference fields */
-			for (i = 0; i < top; i++) {
-				MonoType *ftype;
-
+		/* check for incorrectly aligned or overlapped by a non-object field */
+#ifdef ENABLE_NETCORE	
+		guint8 *layout_check;	
```

Comment:
What was the problem with the old check code ? Isn't changing the #if 0 into #ifdef ENABLE_NETCORE enough ?

Reply:
No, I have to change klass->has_references to has_references, I have to change MONO_TYPE_IS_REFERENCE(field->type) to type_has_references (klass, ftype), and this old check code does 2 loops that aren't necessary in the new implementation.